Feat/add support for field config as array

anaivanovska opened the pull request.

Commit:
feat(grid): Add support for field as string array

If we have a nested objects, then the field
should be defined as path (array of string).
This way we will make sure that the raw value
will be properly calculated.
Also, with this improvement we can remove some value getters from the configuration where path was used
in order to get the value.

Commit:
feat(grid): Adjust tests

Commit:
feat(grid): Add example in the demo app

hdimitrieski commented on projects/data-grid/src/lib/models/data-grid-column.ts:
```diff
@@ -58,14 +59,16 @@ const isFilterDefinedAndApplied = R.allPass([columnFilterDefined, R.compose(filt
-export const getRawValueResolver = ({field}: DataGridColumnWithId): ColumnValueGetter => R.prop(field);
+export const getRawValueResolver = ({field}: DataGridColumnWithId): ColumnValueGetter => isString(field)
+  ? R.prop(field)
+  : R.path(field);
```
What happens if the field or its parent is null or undefined? Can you add a test case?

anaivanovska replied:
In that case the raw value resolver will return 'undefined'.

Commit:
feat(grid): Add test case for not defined fields

Commit:
feat(grid): Rollback column resize calculations

Commit:
feat(grid): Use element width for position calculations

On the components ngrx-grid-header-item and
ngrx-dynamic-grid-header-item we do not
apply the defined width.
This width is applied on upper level, on parent
components. In order to make sure that
the right column width will be used, and that the test
will not depended on the screen where it is executed
we will use elementWidth instead of column.width.
